lib/tower.ex

```diff
@@ -573,7 +573,30 @@ defmodule Tower do
     Application.fetch_env!(:tower, :reporters)
   end
 
-  defp async(fun) do
+  def async(fun) do
+    in_unlinked_task(fn ->
+      try do
+        fun.()
+      rescue
+        exception ->
+          in_unlinked_task(fn ->
+            raise ReportEventError,
+              reporter: reporter_from_stacktrace(__STACKTRACE__),
```

Comment:
The deduced reporter module from this function might not match the configured reporter if the `Tower.async` in the reporter was called from a internal module or error raised from internal module.